infra/cifuzz/filestore/github_actions/__init__.py

```diff
@@ -12,15 +12,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Implementation of a filestore using Github actions artifacts."""
-import os
 import logging
+import os
+import shutil
+import tarfile
+import tempfile
 
 import http_utils
 import filestore
 from filestore.github_actions import github_api
 from third_party.github_actions_toolkit.artifact import artifact_client
 
 
+def tar_directory(directory, archive_path):
+  """Tars a |directory| and returns the path of the result. The path will be
+  |archive_path| with a .tar suffix. |archive_path| should not end in .tar."""
+  parent_directory = os.path.dirname(os.path.abspath(directory))
+  basename = os.path.basename(directory)
+  shutil.make_archive(archive_path,
+                      'tar',
+                      root_dir=parent_directory,
+                      base_dir=basename)
+  archive_path += '.tar'
```

Comment:
It would be better/easier for for the caller to do this. It's strange that the caller is expected to construct a path without the extension and then get one back with it.

Reply:
I've fixed this. The reason tar_directory had this ugly behavior is because `shutil.make_archive` did this. I've changed things so this is no longer a problem.